packages/syft/src/syft/serde/json_serde.py

```diff
@@ -0,0 +1,457 @@
+
+    annotation = _unwrap_type_annotation(annotation)
+
+    if not _is_serializable_mapping(annotation):
+        raise ValueError(f"Cannot deserialize {annotation} from JSON")
+
+    _, value_type = get_args(annotation)
+    return {k: deserialize_json(v, value_type) for k, v in value.items()}
+
+
+def _serialize_to_json_bytes(obj: Any) -> str:
+    obj_bytes = sy.serialize(obj, to_bytes=True)
+    return base64.b64encode(obj_bytes).decode("utf-8")
+
+
+def _deserialize_from_json_bytes(obj: str) -> Any:
+    obj_bytes = base64.b64decode(obj)
+    return sy.deserialize(obj_bytes, from_bytes=True)
+
+
+def serialize_json(value: Any, annotation: Any = None, validate: bool = True) -> Json:
+    """
+    Serialize a value to a JSON-serializable object, using the schema defined by the
+    provided annotation.
+
+    Serialization is always done according to the annotation, as the same annotation
+    is used for deserialization. If the annotation is not provided or is ambiguous,
+    the JSON serialization will fall back to serializing bytes. Examples:
+    - int, `list[int]` are strictly typed
+    - `str | int`, `list`, `list[str | int]`, `list[Any]` are ambiguous and serialized to bytes
+    - Optional types (like int | None) are serialized to the not-None type
+
+    The function chooses the appropriate serialization method in the following order:
+    1. Method registered in `JSON_SERDE_REGISTRY` for the annotation type.
+    2. Pydantic model serialization, including all `SyftObjects`.
+    3. Iterable serialization, if the annotation is a strict iterable (e.g., `list[int]`).
+    4. Mapping serialization, if the annotation is a strictly typed mapping with string keys.
+    5. Serialize the object to bytes and encode it as base64.
+
+    Args:
+        value (Any): Value to serialize.
+        annotation (Any, optional): Type annotation for the value. Defaults to None.
+
+    Returns:
+        Json: JSON-serializable object.
+    """
+    if annotation is None:
+        annotation = type(value)
+
+    if value is None:
+        return None
+
+    # Remove None type from annotation if it is present.
+    annotation = _unwrap_type_annotation(annotation)
+
+    if annotation in JSON_SERDE_REGISTRY:
+        result = JSON_SERDE_REGISTRY[annotation].serialize(value)
+    elif _annotation_issubclass(annotation, pydantic.BaseModel):
+        result = _serialize_pydantic_to_json(value)
+    elif _annotation_issubclass(annotation, Enum):
+        result = value.name
+
+    # JSON recursive types
+    # only strictly annotated iterables and mappings are supported
+    # example: list[int] is supported, but not list[int | str]
+    elif _is_serializable_iterable(annotation):
+        result = _serialize_iterable_to_json(value, annotation)
+    elif _is_serializable_mapping(annotation):
+        result = _serialize_mapping_to_json(value, annotation)
+    else:
+        result = _serialize_to_json_bytes(value)
```

Comment:
why do we need this? We also still store the bytes separately I think? would we ever query on the bytes?

Reply:
Many types in syft are not serializable to json, either because they are custom types or because they are untypedd (and cannot be serialize back to the python obj). they are stored inside the json as bytestring.

for example: UserCodeStatusCollection is not serializable because the keys are some custom object, so the db would look like:

```
{
    "__canonical_name__": "UserCodeStatusCollection",
    "__version__": 1,
    "status_dict": <bytestring>,
}
```

Reply:
yes, but isnt that redundant?

Reply:
There is no other place we save the full blob, the json in Table.fields is the only data we're saving in the db